app/src/main/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModel.kt

```diff
@@ -150,4 +206,27 @@ class LoginEmailViewModel @Inject constructor(
     fun onProxyPasswordChange(newText: TextFieldValue) {
         loginState = loginState.copy(proxyPassword = newText).updateEmailLoginEnabled()
     }
+
+    fun onCodeChange(newValue: CodeFieldValue) {
+        secondFactorVerificationCodeState = secondFactorVerificationCodeState.copy(code = newValue, isCurrentCodeInvalid = false)
+        if (newValue.isFullyFilled) {
+            login()
+        }
+    }
+
+    fun onCodeVerificationBackPress() {
+        secondFactorVerificationCodeState = secondFactorVerificationCodeState.copy(
+            code = CodeFieldValue(TextFieldValue(""), false),
+            isCodeSent = false,
+            emailUsed = "",
+        )
+    }
+
+    fun onCodeResend() {
+        viewModelScope.launch {
+            resolveCurrentAuthScope()?.let {
+                request2FACode(it)
+            }
+        }
+    }
```

Comment:
The `secondFactorVerificationCodeState` and these three functions are the ones I want to extract to an interface, so the 2FA composable can be reused.